tests/test_filters.py

```diff
@@ -690,3 +772,96 @@ def test_flate_decode__not_rectangular(caplog):
     expected = get_data_from_url(url, name=name)
     assert actual_image.getvalue() == expected
     assert caplog.messages == ["Image data is not rectangular. Adding padding."]
+
+
+def test_main_decode_brotli_without_brotli_installed_subprocess(tmp_path):
+    """Test decode_stream_data raises ImportError via subprocess if brotli is not installed."""
+    original_data = b"some data to be compressed with brotli"
+    # We need brotli here in the main process to create the test data
+    try:
+        import brotli
+
+        compressed_data = brotli.compress(original_data)
+    except ImportError:
+        pytest.skip("brotli library not installed in the main test environment")
+
+    script = f"""
+import sys
+import pytest
+from pypdf import filters
+from pypdf.generic import DictionaryObject, NameObject
+
+# Simulate brotli not being installed
+sys.modules["brotli"] = None
+# Need to reload filters to make the None effective inside the module
+import importlib
+import pypdf.filters
+importlib.reload(pypdf.filters)
+
+# Simulate a stream dictionary indicating BrotliDecode
+stream = DictionaryObject()
+stream[NameObject("/Filter")] = NameObject("/BrotliDecode")
+# Pass compressed data as hex to avoid encoding issues in the script string
+stream._data = bytes.fromhex('{compressed_data.hex()}')
+
+# Call the main decode function and expect an error
+with pytest.raises(ImportError) as exc_info:
+    filters.decode_stream_data(stream)
+
+assert "Brotli library not installed. Required for BrotliDecode filter." in str(exc_info.value)
+print("Test finished successfully: main decode without brotli") # Add print to confirm script completion
+"""
+    _run_script_without_brotli(tmp_path, script)
+
+
+# Renamed from test_main_decode_brotli
+def test_main_decode_brotli_installed():
+    """Test the main decode function with Brotli filter using a real PDF."""
+    if importlib.util.find_spec("brotli") is None:
+        pytest.skip("brotli library not installed")
+
+    # Use the prototype PDF provided by PDF Association
+    pdf_path = RESOURCE_ROOT / "brotli-test-pdfs" / "minimal-brotli-compressed.pdf"
+    if not pdf_path.exists():
+        pytest.skip(f"Brotli test PDF not found at {pdf_path}")
+
+    reader = PdfReader(pdf_path)
```

Comment:
Do we really need this large code setup? Isn't it sufficient to just run `page.extract_text()` for each page and make sure nothing fails and returns the correct text?